File: SchedulingFront-OLD/src/app/models/schedule.ts
import { BasePaging } from '../common/models/basePaging';
import { IDayDetail } from '../common/components/calendar/base/dayDetail';
import { Employee } from './employee';
import { Customer } from './customer';
import { CalendarPaging } from '../common/models/calendarPaging';

export class Schedule implements IDayDetail {
    id: number;
    scheduleId: number;
    customerId: number;
    customer: Customer = new Customer();
    customerName: string;
    phoneNumber: string;
    date: Date;
    employeeId: number;
    employee: Employee;
    employees: string;
    bindToEmployee: boolean;
    constructor() {
        this.scheduleId = null;
        this.customer = null;
        this.phoneNumber = null;
        this.date = null;
        this.customerName = null;
        this.employees = null;
    }

    static init(schedule: Schedule) {
        const sch = new Schedule();
        Object.keys(sch).forEach(element => {
            sch[element] = schedule[element];
        });
        sch.id = schedule.scheduleId;
        return sch;
    }
}

// export class ScheduleMap {
//     schedule: Schedule;
// }

export class SchedulePaging extends CalendarPaging {
    dateFrom: Date;
    dateTo: Date;
    filter: any;
    employees = new Array<number>();
    organizationUnits = new Array<number>();

    setCalendarDate(calendarPaging: CalendarPaging) {
        this.dateFrom = calendarPaging.dateFrom;
        this.dateTo = calendarPaging.dateTo;
    }
}


export class ScheduleOnDayPaging extends BasePaging {
    date: Date;
}

File: SchedulingAPI/Scheduling.Contract/Models/Schedule.cs
using System;
using System.Collections.Generic;

namespace Scheduling.Contract.Models
{
    public class Schedule
    {
        public long? ScheduleId { get; set; }
        public DateTime Date { get; set; }
        public long CustomerId { get; set; }
        public string Employees { get; set; }
        public long? Id => ScheduleId;
        public Customer Customer { get; set; } = new Customer();
    }
    public class SchedulePaging
    {
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
        public List<int> Employees = new List<int>();
        public List<int> OrganizationUnits = new List<int>();
        public int ScheduleGroupId { get; set; }
    }
}
